weldx/asdf/util.py

```diff
@@ -509,7 +509,11 @@ def get_highest_tag_version(
     if not tags:  # no match found
         return None
 
-    tags.sort(key=LooseVersion)
+    def _get_version(tag_name: str):
+        # we assume, that the version of the tag is separated by a right-most '-' char.
+        return Version(tag_name[tag_name.rfind("-") + 1 :])
+
+    tags.sort(key=_get_version)
```

Comment:
```suggestion
    tags.sort(key=packaging.version.parse)
```
This should work but seems to use `LegacyVersion` in the back so if we want to avoid that your solution could also be `tags.sort(key=lambda x: Version(x.rpartition("-")[-1]))`

Reply:
It does not use LegacyVersion, the parse function does, if the version format is not pep440 compatible. If we do not omit the tag uri, parse would fall back to LegacyVersion, triggering another DeprecationWarning.